OddSummation.asm

```diff
@@ -0,0 +1,58 @@
+# COMP3410
+# Author: Miguel Betances
+# Assignment: PA[1]
+# Date: Feb/11/2015
+
+# Turn in one .asm file per assignment component
+# Remember to submit it as a pull request to the GitHub repo for the assignment
+
+# OddSummation.asm
+# Input: none
+# Output: Print odd numbers summation from 1-15
+
+##########################################################
+# .data segment. Constant and variable definitions go here.
+##########################################################
+	.data
+
+	head: .asciiz "\nMiguel Betances: Odd Summation\n"
+		storeValue: .word 1	
+		activeValue: .word 0
+		addingValue:.word 1
+		done: .asciiz "\nYour answer is\n"
+							#Done 
+###############################################
+# .text segment. Assembly instructions go here.
+###############################################
+	.text
+	#Display welcome
+	li $v0,4
+	la $a0, head
+	syscall
+loop: 
+	lw $t0,storeValue
+	lw $t1,activeValue
+	li $t2,2		#used to always multiply 2
+	
+	mul $t3,$t0,$t2
+	add $t4,$t1,$t3
+	sw $t4, activeValue
+	lw $t5, addingValue
+	move $t5,$t4
```

Comment:
-2 Inefficient register usage.  If you follow execution 1 step at a time and look at $t2 - $t5 notice that $t2 and $t3 are basically always the same throughout the program, and the same for $t4 and $t5.  You could also limit the number of variables you're using.  You always load the value in storeValue but you never put anyting back in storeValue, so its a constant and could be stored in a register.